src/main/java/teammates/logic/core/StudentsLogic.java

```diff
@@ -629,15 +627,26 @@ private StudentEnrollDetails enrollStudent(StudentAttributes validStudentAttribu
     }
 
     private List<String> getInvalidityInfoInDuplicatedEmail(String email,
-            List<String> studentEmailList, String[] linesArray) {
+            List<StudentAttributes> studentList, String[] linesArray) {
         List<String> info = new ArrayList<>();
-        info.add("Same email address as the student in line \"" + linesArray[studentEmailList.indexOf(email) + 1] + "\"");
+        int index;
+        for (index = 0; index < studentList.size(); index++) {
+            if (studentList.get(index).email.equalsIgnoreCase(email)) {
+                break;
+            }
+        }
+        info.add("Same email address as the student in line \"" + linesArray[index + 1] + "\"");
         return info;
     }
 
     private boolean isStudentEmailDuplicated(String email,
-            List<String> studentEmailList) {
-        return studentEmailList.contains(email);
+            List<StudentAttributes> studentList) {
+        for (StudentAttributes studentInfo : studentList) {
+            if (studentInfo.email.equalsIgnoreCase(email)) {
+                return true;
+            }
+        }
+        return false;
```

Comment:
I noticed that the logic here is quite similar to `getInvalidityInfoInDuplicatedEmail`. 

@whipermr5 do you think it's good if we remove `isStudentEmailDuplicated` and replace it with something like `getDuplicateIndex(String email)`, which returns `-1` if `email` isn't a duplicate and returns the corresponding index if it's a duplicate. That way, we do not have duplicate logic.